cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -489,8 +490,25 @@ def _get_volume_stats(self):
                 summary = datastore["summary"]
 
                 pool_state = 'down'
+                pool_down_reason = 'Datastore not usable'
                 if self.ds_sel.is_datastore_usable(summary):
                     pool_state = 'up'
+                    pool_down_reason = 'up'
+
+                # make sure the datastore isn't in red flag alert!
+                if 'alarms' in datastore and datastore['alarms']:
+                    for alarm in datastore['alarms']:
+                        if alarm[0]['overallStatus'] not in ('red', 'yellow'):
+                            alarm_info = self.volumeops.get_datastore_alarm(
+                                alarm[0]['alarm']
+                            )
+                            if ('volume capacity threshold exceeded' in
+                                    alarm_info['info']['description']):
+                                # vCenter will fail to create volumes here
+                                pool_state = 'down'
+                                pool_down_reason = (
+                                    alarm_info['info']['description']
```

Comment:
I don't think this is how it works. You would retrieve `['info']` from the dict we built in `get_object_properties_dict`, but `info` would then be a `vim.alarm.AlarmInfo` object, which needs access via `.description`. Therefore, if we don't change this as suggested to fetching only `['info.description']`, this would have to be `['info'].description`.